raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2247,7 +2346,18 @@ public synchronized OptionalLong nextExpectedOffset() {
                     return OptionalLong.empty();
                 }
             } else {
-                return OptionalLong.of(lastAckedOffset);
+                return OptionalLong.of(lastAckedEndOffset);
+            }
+        }
+
+        /**
+         * This API is used when the Listener needs to be notified of a new Snapshot. This happens
+         * when the context last acked end offset is less that then log start offset.
+         */
+        public void fireHandleSnapshot(long logStartOffset) {
```

Comment:
This is a temporary signature. This will change when we implement the API for loading snapshots. Tracking is work in this issue: https://issues.apache.org/jira/browse/KAFKA-12154